cdap-docs/developers-manual/source/building-blocks/metadata.rst

```diff
@@ -0,0 +1,44 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _metadata:
+
+========
+Metadata
+========
+
+.. include:: ../../../_common/_includes/include-v320-beta-metadata.rst
+
+Metadata |---| consisting of **properties** (a list of key-value pairs) or **tags** (a
+list of keys) |---| can be set for both data and applications.
+
+Using the CDAP :ref:`Metadata HTTP RESTful API <http-restful-api-metadata>` to set,
+retrieve, and delete the metadata annotations of applications, datasets, streams, and
+other elements in CDAP.
+
+Metadata keys and tags must conform to the CDAP
+:ref:`supported characters <supported-characters>`, and are limited to 50 characters in
+length. Metadata values are limited to 10K bytes in length.
+
+Metadata can be used to tag different CDAP components so that they are easily identifiable
+and managed. You can tag a datasets as *experimental* or an application as *production*.
+
+Metadata can be **searched**, either to find entities with certain properties or values, or to
+find those with a particular tag.
+
+.. _metadata-lineage:
+
+A metadata **lineage** can be retrieved for dataset and stream entities. A lineage shows
```

Comment:
Lineage is a separate concept, which I think deserves its own subtitle, if not more. Also, s/metadata lineage/lineage/